crates/libs/windows/src/Windows/Win32/Graphics/Direct3D12/mod.rs

```diff
@@ -48,14 +48,14 @@ pub unsafe fn D3D12CreateVersionedRootSignatureDeserializer(psrcdata: *const ::c
 }
 #[doc = "*Required features: 'Win32_Graphics_Direct3D12'*"]
 #[inline]
-pub unsafe fn D3D12EnableExperimentalFeatures(numfeatures: u32, piids: *const ::windows::core::GUID, pconfigurationstructs: *const ::core::ffi::c_void, pconfigurationstructsizes: *const u32) -> ::windows::core::Result<()> {
+pub unsafe fn D3D12EnableExperimentalFeatures(piids: &[::windows::core::GUID], pconfigurationstructs: &[u8], pconfigurationstructsizes: &[u32]) -> ::windows::core::Result<()> {
     #[cfg(windows)]
     {
         #[link(name = "windows")]
         extern "system" {
             fn D3D12EnableExperimentalFeatures(numfeatures: u32, piids: *const ::windows::core::GUID, pconfigurationstructs: *const ::core::ffi::c_void, pconfigurationstructsizes: *const u32) -> ::windows::core::HRESULT;
         }
-        D3D12EnableExperimentalFeatures(::core::mem::transmute(numfeatures), ::core::mem::transmute(piids), ::core::mem::transmute(pconfigurationstructs), ::core::mem::transmute(pconfigurationstructsizes)).ok()
+        D3D12EnableExperimentalFeatures(pconfigurationstructsizes.len() as _, ::core::mem::transmute(::windows::core::as_ptr_or_null(piids)), ::core::mem::transmute(::windows::core::as_ptr_or_null(pconfigurationstructs)), ::core::mem::transmute(::windows::core::as_ptr_or_null(pconfigurationstructsizes))).ok()
```

Comment:
The three arrays passed in to this all have to be the same length.  As it's also valid to pass in null for the configuration structs, relying on that to get the length could be a problem.  This pIIDs parameter is non-optional and should really be authoritative about the size of the arrays.

Reply:
Good catch! If there's a shared length parameter, I'll have to look for a "winner" and the non-optional parameter can be the winner by default.

Reply:
I updated the algorithm to detect shared length parameters and to avoid the transformation if there any optional parameters among them. That's probably more conservative than we need to be, but we can tweak it later if it becomes an issue.

Reply:
This function is no longer transformed.